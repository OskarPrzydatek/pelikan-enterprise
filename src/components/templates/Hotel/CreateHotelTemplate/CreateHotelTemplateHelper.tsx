import { useForm } from 'react-hook-form';

import { CreateHotelTemplate } from '~/components/templates';
import { IHotel } from '~/models';
import { StyledComponentsProvider } from '~/providers';

interface ICreateHotelTemplateHelper {
  onSubmitCreateHotel: () => void;
}

export const CreateHotelTemplateHelper: React.FC<
  ICreateHotelTemplateHelper
> = ({ onSubmitCreateHotel }: ICreateHotelTemplateHelper) => {
  const methods = useForm<IHotel>();

  return (
    <StyledComponentsProvider>
      <CreateHotelTemplate
        createHotelFormMethods={methods}
        onSubmitCreateHotel={onSubmitCreateHotel}
      />
    </StyledComponentsProvider>
  );
};
