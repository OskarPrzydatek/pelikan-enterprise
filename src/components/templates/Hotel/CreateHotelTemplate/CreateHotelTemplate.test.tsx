import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { CreateHotelTemplateHelper } from './CreateHotelTemplateHelper';

const mockOnSubmitCreateHotel = vi.fn();

describe('CreateHotelTemplate', () => {
  test('component snapshot', () => {
    const view = render(
      <CreateHotelTemplateHelper
        onSubmitCreateHotel={mockOnSubmitCreateHotel}
      />
    );
    expect(view).toMatchSnapshot();
  });
});
