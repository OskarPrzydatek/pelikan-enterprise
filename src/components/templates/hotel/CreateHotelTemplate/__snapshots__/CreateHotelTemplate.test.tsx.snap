// Vitest Snapshot v1

exports[`CreateHotelTemplate > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iAEawV eSwvqh"
      >
        <form
          class="sc-pyfCe rsgVC"
        >
          <h2
            class="sc-dkrFOg kaKqDq"
          >
            Dodaj hotel
          </h2>
          <div
            class="sc-gKPRtg gOHhkn"
          >
            <div
              class="sc-idXgbr doOTQv"
            >
              <div
                class="sc-dIfARi fkTxxa"
              >
                <input
                  class="sc-hHTYSt fAxxrI"
                  data-testid="create-hotel-name"
                  id="name"
                  name="name"
                  placeholder="Nazwa hotelu"
                />
              </div>
            </div>
            <div
              class="sc-idXgbr doOTQv"
            >
              <div
                class="sc-dIfARi fkTxxa"
              >
                <input
                  class="sc-hHTYSt fAxxrI"
                  data-testid="create-hotel-address"
                  id="address"
                  name="address"
                  placeholder="Adres"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-idXgbr doOTQv"
          >
            <div
              class="sc-dIfARi fkTxxa"
            >
              <input
                class="sc-hHTYSt fAxxrI"
                data-testid="create-hotel-stars-rating"
                id="starRating"
                name="starRating"
                placeholder="Ilość gwiazdek"
              />
            </div>
          </div>
          <div
            class="sc-hBxehG dvBsjd"
          >
            <div
              class="sc-fnGiBr rTdrx"
            >
              <textarea
                class="sc-bjfHbI bpDBhG"
                data-testid="create-hotel-description"
                id="description"
                name="description"
                placeholder="Opis"
              />
            </div>
          </div>
          <input
            class="sc-jrcTuL iZHkcL"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iAEawV eSwvqh"
    >
      <form
        class="sc-pyfCe rsgVC"
      >
        <h2
          class="sc-dkrFOg kaKqDq"
        >
          Dodaj hotel
        </h2>
        <div
          class="sc-gKPRtg gOHhkn"
        >
          <div
            class="sc-idXgbr doOTQv"
          >
            <div
              class="sc-dIfARi fkTxxa"
            >
              <input
                class="sc-hHTYSt fAxxrI"
                data-testid="create-hotel-name"
                id="name"
                name="name"
                placeholder="Nazwa hotelu"
              />
            </div>
          </div>
          <div
            class="sc-idXgbr doOTQv"
          >
            <div
              class="sc-dIfARi fkTxxa"
            >
              <input
                class="sc-hHTYSt fAxxrI"
                data-testid="create-hotel-address"
                id="address"
                name="address"
                placeholder="Adres"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-idXgbr doOTQv"
        >
          <div
            class="sc-dIfARi fkTxxa"
          >
            <input
              class="sc-hHTYSt fAxxrI"
              data-testid="create-hotel-stars-rating"
              id="starRating"
              name="starRating"
              placeholder="Ilość gwiazdek"
            />
          </div>
        </div>
        <div
          class="sc-hBxehG dvBsjd"
        >
          <div
            class="sc-fnGiBr rTdrx"
          >
            <textarea
              class="sc-bjfHbI bpDBhG"
              data-testid="create-hotel-description"
              id="description"
              name="description"
              placeholder="Opis"
            />
          </div>
        </div>
        <input
          class="sc-jrcTuL iZHkcL"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
